src/graphnet/models/components/embedding.py

```diff
@@ -0,0 +1,27 @@
+"""Classes for performing embedding of input data."""
+import torch
+
+
+class SinusoidalPosEmb(torch.nn.Module):
+    """Sinusoidal positional embedding layer."""
+
+    def __init__(self, dim: int = 16, M: int = 10000) -> None:
+        """Construct `SinusoidalPosEmb`.
+
+        Args:
```

Comment:
I think a nice fat doc string here would be good. It's a bit hard to read this code and understand what it does and why this is motivated. Perhaps you could spell it out here and point to the kaggle paper for details?